### ISSUE TITLE
artran needs to update SAL record upon increases in total sale

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#353]
_Imported from trac ticket 353.
Created by **rneville**
Opened in trac: Oct 20, 2010, 4:07:29 PM
Last modified in trac: Nov 5, 2010, 12:54:33 PM_
***

**Comment by kentbower** [comment]
_**rneville** trac status change on Oct 25, 2010, 3:01:08 PM:_ **closed**

**Comment by kentbower** [comment]
_**kbower** trac comment on Nov 2, 2010, 11:31:50 AM:_

Please investigate these issues:
 * I suspect `DBSession.query(ArTran).filter_by(adjinvoice=self.orderid` is incorrect.  Shouldn't it be .filter_by(orderid=self.orderid ???
 * I suspect our logic oversimplifies when another cust_cd has transactions against the order.  It is possible there are two 'SAL' artrans records in that case?  Should we be filtering against customerid as well?  Please investigate.
 * We are not checking the type of order.
```
Kent,

Good morning, only 'SAL' orders will have a 'SAL' AR_TRN TRN_TP_CD after
they are saved.

Thanks,
 
Wayne Liburd
```

**Comment by kentbower** [comment]
_**kbower** trac status change on Nov 2, 2010, 11:31:50 AM:_ **reopened**

**Comment by kentbower** [comment]
_**kbower** trac comment on Nov 2, 2010, 2:36:08 PM:_

(In [[883]](../commit/4daf7ded1a8386bf5b6933a9ee113956e4657fd1))
 * Remove all invocations of .first() as it is never correct unless we truly intend to grab a random record.  In all cases, .one() or much better .get() is what is really intended.  .get() won't even make a trip to the database if the object is already loaded.
   * Also prevent future invocations of .first().  Closes [#418](418)
 * fix problem caused in dataload when we reference a related object that hasn't been loaded yet during a session.flush().  In this case, the commited (and therefore incorrect or nonexistent) value of the foreign key is referenced, causing the related object to not load
 * fix a bunch of logic in the financing code, but see [#425](425) and #353

**Comment by kentbower** [comment]
_**rneville** trac comment on Nov 5, 2010, 12:54:33 PM:_

(In [[888]](../commit/2fe85a8c57c1825b6e1bba7560f7875d405cc3c9))
fixes [ticket:425](425) and also fixes #353

**Comment by kentbower** [comment]
_**rneville** trac status change on Nov 5, 2010, 12:54:33 PM:_ **closed**
